File: DesignPatterns/Creational/Singleton/CSharp/Implementation/Singleton.cs
namespace DesignPatterns.Creational.Singleton.CSharp.Implementation
{
    public class Singleton
    {
        private static Singleton _Instance { get; set; }

        private Singleton() { }

        public static Singleton GetInstance()
        {
            if (_Instance == null)
            {
                _Instance = new Singleton();
            }

            return _Instance;
        }
    }
}

File: DesignPatterns/Creational/Singleton/TypeScript/Singleton.ts
export class Singleton {
    private static _instance: Singleton | null = null;

    private constructor() { }

    public static getInstance(): Singleton {
        if (this._instance == null) {
            this._instance = new this();
        }

        return this._instance;
    }
}

// example
export class SingletonUsage {

    constructor() { }

    public someFunction(): void {
        const x: Singleton = Singleton.getInstance();

        // do whatever
    }
}